### PR TITLE
Add user context auth id field

### PR DESCRIPTION
**Pull request by Assaf-Neufeld**

### CODE REVIEW

File: src/Core/Managed/Net45/Extensibility/Implementation/External/UserContextData.cs [hunk 1]
```diff
@@ -77,6 +77,15 @@ public string StoreRegion
         {
             get { return this.tags.GetTagValueOrNull(ContextTagKeys.Keys.UserStoreRegion); }
             set { this.tags.SetStringValueOrRemove(ContextTagKeys.Keys.UserStoreRegion, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the AuthUserId of an application-defined account associated with the user.
+        /// </summary>
+        public string AuthUserId
```
Comment (SergeyKanzhelev) [inline src/Core/Managed/Net45/Extensibility/Implementation/External/UserContextData.cs]: Name is bad. @JoshuaWeber, can you please help with the name here?

**Comment by SergeyKanzhelev** [comment]
there are also UWP and 4.6 projects recently merged by @MeniZalzman . Make sure you've updated them as well

File: src/Core/Managed/Net45/Extensibility/Implementation/External/UserContextData.cs (hunk 1 above)
Comment (Assaf-Neufeld) [inline src/Core/Managed/Net45/Extensibility/Implementation/External/UserContextData.cs]: This is the name of the field in the BOND schema. I can't/shouldnt modify it here

Comment (SergeyKanzhelev) [inline src/Core/Managed/Net45/Extensibility/Implementation/External/UserContextData.cs]: This is a public class we expose to customers and we need to make sure that names are user friendly and well-explained. It doesn't matter what are the names of the actual schema fields

Comment (Assaf-Neufeld) [inline src/Core/Managed/Net45/Extensibility/Implementation/External/UserContextData.cs]: This is the generated class.
I missed updating the public "UserContext" class. I've updated the PR with the additional change.  I named the property "AuthenticatedUserId" - does that sound good to you?

Comment (SergeyKanzhelev) [inline src/Core/Managed/Net45/Extensibility/Implementation/External/UserContextData.cs]: Yes. Makes sense now

**Comment by abaranch** [comment]
:shipit:

**Pull request merged**